hydra-node/src/Hydra/Logging/Monitoring.hs

```diff
@@ -38,19 +43,55 @@ withMonitoring (Just monitoringPort) (Tracer tracer) action = do
 -- | Register all relevant metrics.
 -- Returns an updated `Registry` which is needed to `serveMetrics` or any other form of publication
 -- of metrics, whether push or pull, and a function for updating metrics given some trace event.
-prepareRegistry :: MonadIO m => m (HydraLog tx net -> m (), Registry)
-prepareRegistry = first monitor <$> registerMetrics
+prepareRegistry :: (MonadIO m, MonadSTM m, MonadMonotonicTime m, Tx tx) => TVar m (Map (TxId tx) Time) -> m (HydraLog tx net -> m (), Registry)
+prepareRegistry txMap = first monitor <$> registerMetrics
  where
+  monitor metricsMap (Node (ProcessingEvent _ (NetworkEvent (ReqTx _ tx)))) = do
+    t <- getMonotonicTime
+    atomically $ modifyTVar' txMap (Map.insert (txId tx) t)
```

Comment:
This `txMap` is currently ever growing, right? Might want to `NOTE` this, as this is likely going to be an issue when doing things in production.

Reply:
Ouch, I will fix it.